test/functional/feature_block.py

```diff
@@ -1269,20 +1269,32 @@ def run_test(self):
         self.move_tip(88)
         blocks2 = []
         for i in range(89, LARGE_REORG_SIZE + 89):
-            blocks2.append(self.next_block("alt" + str(i)))
+            blocks2.append(self.next_block("alt" + str(i), version=4))
         self.send_blocks(blocks2, False, force_send=True)
 
         # extend alt chain to trigger re-org
-        block = self.next_block("alt" + str(chain1_tip + 1))
+        block = self.next_block("alt" + str(chain1_tip + 1), version=4)
         self.send_blocks([block], True, timeout=960)
 
         # ... and re-org back to the first chain
         self.move_tip(chain1_tip)
-        block = self.next_block(chain1_tip + 1)
+        block = self.next_block(chain1_tip + 1, version=4)
         self.send_blocks([block], False, force_send=True)
-        block = self.next_block(chain1_tip + 2)
+        block = self.next_block(chain1_tip + 2, version=4)
         self.send_blocks([block], True, timeout=960)
 
+        self.log.info("Reject a block with an invalid block header version")
+        b_v1 = self.next_block('b_v1', version=1)
+        self.sync_blocks([b_v1], success=False, force_send=True, reject_reason='bad-version(0x00000001)')
+
+        self.move_tip(chain1_tip + 2)
+        b_cb34 = self.next_block('b_cb34', version=4)
+        b_cb34.vtx[0].vin[0].scriptSig = b_cb34.vtx[0].vin[0].scriptSig[:-1]
+        b_cb34.vtx[0].rehash()
+        b_cb34.hashMerkleRoot = b_cb34.calc_merkle_root()
+        b_cb34.solve()
+        self.sync_blocks([b_cb34], success=False, reject_reason='bad-cb-height', reconnect=True)
```

Comment:
This is missing some previous back port to allow us to use success and reject_reason

Reply:
did you see this?

Reply:
Oh shoot, I'm seeing this now. Will get on it

Reply:
Are you sure its 11817, it only touches `feature_csv_activation.py`